changelogs/fragments/tarfile_extract_warn.yml

```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- tarfile - handle data filter deprecation warning message for extract and extractall (https://github.com/ansible/ansible/issues/80832).
```

Comment:
I'd also include a changelog entry for ansible-galaxy that specifically mentions the use of the tarfile filter, since that's not obvious unless looking at the changes in this PR.